GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/GVFSUpgradeNagTests.cs

```diff
@@ -0,0 +1,89 @@
+﻿using GVFS.FunctionalTests.FileSystemRunners;
+using GVFS.FunctionalTests.Tools;
+using GVFS.Tests.Should;
+using NUnit.Framework;
+using System;
+using System.IO;
+using System.Linq;
+
+namespace GVFS.FunctionalTests.Tests.EnlistmentPerFixture
+{
+    [TestFixture]
+    [Category(Categories.FullSuiteOnly)]
+    [Category(Categories.Windows)]
+    public class UpgradeNagTests : TestsWithEnlistmentPerFixture
+    {
+        private const string GVFSInstallerName = "SetupGVFS.1.0.18234.1.exe";
+        private const string GitInstallerName = "Git-2.17.1.gvfs.2.5.g2962052-64-bit.exe";
+
+        private string upgradeDirectory;
+        private FileSystemRunner fileSystem;
+
+        public UpgradeNagTests()
+        {
+            this.fileSystem = new SystemIORunner();
+            this.upgradeDirectory = Path.Combine(
+                Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData, Environment.SpecialFolderOption.Create),
+                "GVFS",
+                "GVFS.Upgrade",
+                "Downloads");
+        }
+
+        [TestCase]
+        [Category(Categories.Windows)]
+        public void NoNagWhenUpgradeNotAvailable()
+        {
+            this.EmptyDownloadDirectory();
+
+            ProcessResult result = GitHelpers.InvokeGitAgainstGVFSRepo(
+                this.Enlistment.RepoRoot,
+                "status");
+
+            string.IsNullOrEmpty(result.Errors).ShouldBeTrue();
+        }
+
+        [TestCase]
+        [Category(Categories.Windows)]
+        public void NagWhenUpgradeAvailable()
+        {
+            this.CreateUpgradeInstallers();
+
+            ProcessResult result = GitHelpers.InvokeGitAgainstGVFSRepo(
+                this.Enlistment.RepoRoot,
+                "status");
+
+            result.Errors.ShouldContain(new string[] 
+                {
+                    "A newer version of GVFS is available.",
+                    "Run \"gvfs upgrade\" on \"Command Prompt\" as Administrator to install."
+                });
+
+            this.EmptyDownloadDirectory();
+        }
+
+        private void EmptyDownloadDirectory()
+        {
+            if (Directory.Exists(this.upgradeDirectory))
+            {
+                Directory.Delete(this.upgradeDirectory, recursive: true);

```

Comment:
If someone has installed GVFS on their machine and actually has pending GVFS updates to install, will running the functional tests delete the (real) upgraders?

Reply:
Yes, it would delete previously downloaded upgraders.

Reply:
> Yes, it would delete previously downloaded upgraders.

Would the service eventually re-download them?

Reply:
Yes - there is an upgrade timer in the Service that runs every day.